src/inmanta/server/server.py

```diff
@@ -1570,6 +1571,7 @@ def delete_environment(self, environment_id):
         if env is None:
             return 404, {"message": "The environment with given id does not exist."}
 
+        yield self.agentmanager.stop_agents(env)
```

Comment:
same here. perhaps it would be better to yield on both statements, so they run in parallel. Both can take a while.